src/tests/Interop/MonoAPI/MonoMono/InstallEHCallback.cs

```diff
@@ -5,183 +5,183 @@
+            return 5;
+        }
+        else if (o is SpecialExn)
+            return 0;
+        else {
+            Console.Error.WriteLine ("o was not a SpecialExn, it is {0}", o);
+            return 6;
+        }
+    }
+
+    public class Caller2 {
+        public static bool rethrow_called;
+        public static bool exception_caught;
+        public static bool return_from_inner_managed_callback;
+
+        public static void Setup () {
+            rethrow_called = false;
+            exception_caught = false;
+            return_from_inner_managed_callback = false;
+        }
+
+        public static void RethrowException (uint original_exception) {
+            var e = (Exception) GCHandle.FromIntPtr ((IntPtr) original_exception).Target;
+            rethrow_called = true;
+            System.Runtime.ExceptionServices.ExceptionDispatchInfo.Capture (e).Throw ();
+        }
+
+        [MonoPInvokeCallback (typeof (VoidHandleHandleOutDelegate))]
+        public static void Del2 (uint original_exception, out int exception_handle) {
+            exception_handle = 0;
+            try {
+                RethrowException (original_exception);
+            } catch (Exception ex) {
+                var handle = GCHandle.Alloc (ex, GCHandleType.Normal);
+                exception_handle = GCHandle.ToIntPtr (handle).ToInt32 ();
+                exception_caught = true;
+            }
+            return_from_inner_managed_callback = true;
+        }
+    }
+        
+    public static int test_0_throw_and_raise_exception ()
+    {
+        Caller.Setup ();
+        Caller2.Setup ();
+        VoidVoidDelegate f = new VoidVoidDelegate (Caller.M2);
+        VoidHandleHandleOutDelegate del2 = new VoidHandleHandleOutDelegate (Caller2.Del2);
+        bool outer_managed_callback = false;
+        try {
+            mono_test_setup_ftnptr_eh_callback (f, del2);
+        } catch (Exception e) {
+            outer_managed_callback = true;
+        }
+
+        if (!outer_managed_callback) {
+            Console.Error.WriteLine ("outer managed callback did not throw exception");
+            return 1;
+        }
+        if (!Caller2.rethrow_called) {
+            Console.Error.WriteLine ("exception was not rethrown by eh callback");
+            return 2;
+        }
+        if (!Caller2.exception_caught) {
+            Console.Error.WriteLine ("rethrown exception was not caught");
+            return 3;
+        }
+        if (!Caller2.return_from_inner_managed_callback) {
+            Console.Error.WriteLine ("managed callback called from native eh callback did not return");
+            return 4;
+        }
+
+        mono_test_cleanup_ftptr_eh_callback ();
+        return 0;
+    }
+
+    [Fact]
+    public static void TestEntryPoint ()
+    {
+        MonoAPI.Tests.MonoAPISupport.Setup();
+        int result;
+        result = test_0_setjmp_exn_handler ();
+        Assert.Equal(0, result);
```

Comment:
I suspect that this (and the others) end up being fine (mono test, int equality), but I don't know for sure that these don't end up being run in some configuration that breaks.  But hopefully outerloop (and extra platforms?) is good enough to check that?

Reply:
All I can say to this now is that in general Mono tests are supported by the merged test infrastructure, @jkoritzinsky went to great lengths to make that fully functional and IIRC Mono people were happy about his implementation in terms of efficiency and cleanness.

Reply:
Sorry, I wasn't clear.  I just don't know if this test will end up running in some other configuration (from a quick look, I don't see it being excluded from other runtimes).  However, I meant it more as a "heads up" in case something breaks, not an objection to the PR.

Reply:
If we have concerns, we can use the SkipOnCoreClrAttribute on the test and it will automatically be skipped on non-Mono platforms. Tbh I think that's a better solution than suppressing the test in issues.targets.